integration/failpoint/cmd/cni-bridge-fp/main.go

```diff
@@ -0,0 +1,209 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/containerd/containerd/pkg/failpoint"
+	"github.com/containerd/continuity"
+	"github.com/sirupsen/logrus"
+)
+
+type inheritedPodAnnotations struct {
+	// CNIFailpointControlStateDir is used to specify the location of
+	// failpoint control setting. In that such stateDir, the failpoint
+	// setting is stored in the json file named by
+	// `${K8S_POD_NAMESPACE}-${K8S_POD_NAME}.json`. The detail of json file
+	// is described by FailpointConf.
+	CNIFailpointControlStateDir string `json:"cniFailpointControlStateDir,omitempty"`
+}
+
+// FailpointConf is used to describe cmdAdd/cmdDel/cmdCheck command's failpoint.
+type FailpointConf struct {
+	Add   string `json:"cmdAdd"`
+	Del   string `json:"cmdDel"`
+	Check string `json:"cmdCheck"`
+}
+
+type netConf struct {
+	RuntimeConfig struct {
+		PodAnnotations inheritedPodAnnotations `json:"io.kubernetes.cri.pod-annotations"`
+	} `json:"runtimeConfig,omitempty"`
+}
+
+func main() {
+	stdinData, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		logrus.Fatalf("failed to read stdin: %v", err)
+	}
+
+	var conf netConf
+	if err := json.Unmarshal(stdinData, &conf); err != nil {
+		logrus.Fatalf("failed to parse network configuration: %v", err)
+	}
+
+	cniCmd, ok := os.LookupEnv("CNI_COMMAND")
+	if !ok {
+		logrus.Fatal("required env CNI_COMMAND")
+	}
+
+	cniPath, ok := os.LookupEnv("CNI_PATH")
+	if !ok {
+		logrus.Fatal("required env CNI_PATH")
+	}
+
+	evalFn, err := buildFailpointEval(conf.RuntimeConfig.PodAnnotations.CNIFailpointControlStateDir, cniCmd)
+	if err != nil {
+		logrus.Fatalf("failed to build failpoint evaluate function: %v", err)
+	}
+
+	if err := evalFn(); err != nil {
+		logrus.Fatalf("failpoint: %v", err)
+	}
+
+	cmd := exec.Command(filepath.Join(cniPath, "bridge"))
+	cmd.Stdin = bytes.NewReader(stdinData)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Start(); err != nil {
+		logrus.Fatalf("failed to start bridge cni plugin: %v", err)
+	}
+
+	if err := cmd.Wait(); err != nil {
+		logrus.Fatalf("failed to wait for bridge cni plugin: %v", err)
+	}
+}
+
+// buildFailpointEval will read and update the failpoint setting and then
```

Comment:
What is an example of what it is updated to and how it is used?

Reply:
Will add README.me with example in the folder.

Reply:
Updated. ping @dmcgowan